test/blackbox/common/DDSBlackboxTestsContentFilter.cpp

```diff
@@ -615,6 +641,326 @@ TEST(DDSContentFilter, CorrectlyHandleAliasOtherHeader)
     EXPECT_NE(nullptr, filtered_topic);
 }
 
+/**
+ * @brief PubSubTypeTraits for the PubSubReader and PubSubWriter
+ *
+ * This struct enables instantiation of a PubSubReader and PubSubWriter with a DynamicTypeSupport
+ * for a type equivalent to that of KeyedHelloWorld.idl.
+ * That and the fact that this builder sets the auto_fill_type_object flag to true allow for instance to create a
+ * PubSubReader with a content filter, which right now is not possible to do with the Fast DDS generated
+ * TypeSupport, as it does not contain the TypeObject code.
+ *
+ * To use it, do one of:
+ *   - PubSubReader<fastrtps::types::DynamicPubSubType, DynamicTypeTraits> reader;
+ *   - PubSubWriter<fastrtps::types::DynamicPubSubType, DynamicTypeTraits> writer;
+ */
+struct DynamicTypeTraits
```

Comment:
As this `struct` defines the specific trait methods for a dynamically constructed `KeyedHelloWorld`, it may be interesting to rename it to something similiar `DynamicKeyedHelloworldTypeTraits`. Otherwise, the name can be confusing and one may think that it contains generic traits for a type derived from a DynamicPubSubType . Moreover, perhaps it makes sense to move the struct to an external header file within `types`.
What do you think ?
Thanks @MiguelCompany for helping with those suggestions that helped simplifying.